impeller/renderer/context.h

```diff
@@ -154,6 +154,20 @@ class Context {
   ///
   virtual std::shared_ptr<CommandBuffer> CreateCommandBuffer() const = 0;
 
+  //---------------------------------------------------------------------------
+  /// @brief Indicate the beginning of a presentation cycle.
+  ///
+  ///        This may or may not indicate a new frame, or there may be multiple
+  ///        presents within a single frame due to platform view presence.
+  virtual void DidAcquireSurfaceFrame() const = 0;
```

Comment:
in terms of where does the cache live, we could definitely put it in entity pass. But I'm not sure how we live without the frame concept. I don't think its the case that 1 entity pass = 1 presentation cycle, or is it the case that the current cache is based around frames.

FYI @dnfield , does you AIKS canvas patch cause us to create multiple entity passes per frame (non-platform view)? or are impeller picutres always combined into a single entity pass?

Reply:
Well, we can't store it in entity pass - but we could store it on the content context and access via entity pass.

Reply:
> is the context the HAL?

Yeah, `impeller::Context` is the central resource of the HAL, since you use it to spawn all GPU resources and behaviors. By HAL I mean pretty much everything under `impeller/renderer`, or our equivalent or `gfx_hal` or WebGPU.

> and what is the application?

Renderers built atop the HAL. `EntityPass`, Flutter GPU renderers, and my stupid side projects 😛.

> But I'm not sure how we live without the frame concept. I don't think its the case that 1 entity pass = 1 presentation cycle, or is it the case that the current cache is based around frames.

One call to `EntityPass::Render()` should happen per-platform view per-frame.

> does you AIKS canvas patch cause us to create multiple entity passes per frame (non-platform view)?

Oh, do you mean the `DrawPicture` stuff? `DrawPicture` merges a cloned `EntityPass` from a picture into another and won't result in `EntityPass::Render()` being called more than once per-platform view per-frame.

Reply:
Ah, you probably mean https://github.com/flutter/engine/pull/44248.

Reply:
SGTM, I'll move the allocation cache to a structure on the content context. I think I'll adjust the render target constructors a bit so that they also accept an allocator - and I can wrap the context allocator in a caching content_context allocator?

Reply:
TBH the `RenderTarget::CreateOffscreen[MSAA]` static factories wound up being super Entities-specific in the end and belong in something like `ContentContext` anyhow. Moving those factories up into Entities and passing in some optional Entities-specific cache through `EntityPass CreateRenderTarget` seems like a potentially good route.